examples/optim.jl

```diff
@@ -0,0 +1,299 @@
+# # Optimizing agent-based models
+
+# Agent-based models (ABMs) are computationally expensive and can have many parameters. Sometimes we need to fine-tune a model's parameters to a specific outcome. Brute-force algorithms can take too long for testing each parameter setting. Even if it was feasbile to run the model for every parameter setting, it would not be enough because ABMs are stochastic and the effect of a parameter setting should be derived from running the model several times and taking its average behavior.
```

Comment:
They don't need to be computationally expensive _per se_. 

Agent-based models (ABMs) can have many parameters, and sometimes we need to fine-tune a model's parameters to a specific outcome. A model with hundreds of parameters (or more) makes brute-force parameter scanning unfeasible. On top of that, ABMs are stochastic and the effect of setting just one parameter should be derived from running the model several times and assigning the average behavior.

Reply:
Models with much fewer parameters can take too long to explore the parameter space.

Reply:
Size of the parameter space of a simple model with 10 parameters and 10 states per parameter is 10^10. Even if each simulation takes only one second, exploring the whole space would take more than 300 years.

Reply:
Oh, certainly. Just that they are not *computationally expensive* as a fundamental aspect of them. We have plenty of examples that show that this is not the case.

Reply:
Well, comparing them with analytical solutions, they are almost always more expensive, aren't they?

Reply:
If you have one parameter `x` and want to test `for x in 0:10`.

Reply:
> What does "state per parameter" mean?

It means that a parameter can take 10 different values, for example.

Reply:
Oooooh that is what you mean. So "states" = "possible values"

Reply:
I think using `state` here is kind of confusing, because "state space" is used (in the lingo of dynamical systems) to describe the entire state space (which for the SIR model is 1-dimensional set of three values :S, :I, :R) .

Reply:
Fair enough. Parameter state for parameter categories is commonly used in the context of graphical models. Tim understood me because he is also reading about such models.

Reply:
If you feel "state" is more appropriate then feel free to use it instead.